packages/core/src/clients/middleware/retry/retryInfoMiddleware.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { HttpRequest, HttpResponse, Middleware } from '../../types';
+
+import { RetryOptions } from './retryMiddleware';
+import { DEFAULT_RETRY_ATTEMPTS, SDK_REQUEST_HEADER } from './constants';
+
+export type RetryInfoMiddlewareOption = Pick<RetryOptions, 'maxAttempts'>;
+
+/**
+ * Middleware injects `amz-sdk-request` header to indicate the retry states that the time an HTTP request is made.
```

Comment:
```suggestion
 * Middleware injects `amz-sdk-request` header to indicate the retry state at the time an HTTP request is made.
```
Not sure if this suggestion is correct, but original could be more clear.